.github/workflows/publish-test-image.yml

```diff
@@ -0,0 +1,80 @@
+name: build-test-image
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        required: true
+        default: build
+    secrets:
+      aws-role-arn:
+        required: true
+        description: >
+          The Amazon Resource Name (ARN) of the role to assume. Use the provided credentials to assume an IAM role and
+          configure the Actions environment with the assumed role credentials rather than with the provided credentials.
+      ecr-repository:
+        required: true
+        description: >
+          The URL of the ECR repository to publish the test image to.
+
+concurrency:
+  group: publish-test-image-${{ github.head_ref || github.ref_name }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+  id-token: write
+  packages: read
+
+jobs:
+  build-test-image:
+    name: Build Test Image
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.environment }}
+    permissions:
+      contents: read
+      id-token: write
+      packages: read
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v5
+
+      - name: Assume AWS Role
+        uses: aws-actions/configure-aws-credentials@7474bc4690e29a8392af63c5b98e7449536d5c3a # v4.3.1
+        with:
+          role-to-assume: ${{ secrets.aws-role-arn }}
+          aws-region: "eu-west-2"
+
+      - name: Login to Amazon ECR
+        id: login-to-ecr
+        uses: aws-actions/amazon-ecr-login@062b18b96a7aff071d4dc91bc00c4c1a7945b076 # v2.0.1
+
+      - name: Setup node
```

Comment:
This step may seem superfluous, but it is required to generate an `.npmrc` file that gets mounted into the docker image as a build secret.